### PR TITLE
feat: support milvus

### PR DESCRIPTION
**Pull request by jupyterjazz**
Taking over from here https://github.com/docarray/docarray/pull/1666

- [x] Find (vector search)
- [x] Filter
- [x] Get/Del
- [x] Hybrid Search (find+filter)
- [x] Subindex
- [x] Tests

Functionality-wise everything's ready, only thing left is to make the execution faster (milvus job takes 13 mins on CI)

Docs will be done in a separate PR

Note:
I had to bump protobuf and tensorflow versions

### CODE REVIEW

File: docarray/index/backends/milvus.py [hunk 1]
```diff
@@ -0,0 +1,841 @@
...
+        type_map = {
+            int: DataType.INT64,
+            float: DataType.FLOAT,
+            str: DataType.VARCHAR,
+            bytes: DataType.VARCHAR,
+            np.ndarray: DataType.FLOAT_VECTOR,
+            list: DataType.FLOAT_VECTOR,
+            AnyTensor: DataType.FLOAT_VECTOR,
+            AbstractTensor: DataType.FLOAT_VECTOR,
+        }
```
Comment (samsja) [inline docarray/index/backends/milvus.py:192]: We can move this dict away from the function imo

**Review by samsja — COMMENTED** [review]
added one minor comments

File: docarray/index/backends/milvus.py (hunk 1 above)
Comment (jupyterjazz) [reply docarray/index/backends/milvus.py:192]: hmm but all other backends have it inside

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -198,7 +198,7 @@ jobs:
           python -m pip install --upgrade pip
           python -m pip install poetry
           poetry install --all-extras
-          poetry run pip install protobuf==3.19.0 # we check that we support 3.19
+          poetry run pip install protobuf==3.20.0 # we check that we support 3.19
```
Comment (samsja) [inline .github/workflows/ci.yml:201]:
```suggestion
          poetry run pip install protobuf==3.20.0 # we check that we support 3.20
```

**Review by samsja — APPROVED** [review]

**Review by JoanFM — APPROVED** [review]

**Pull request merged**